src/compiler/crystal/codegen/debug.cr

```diff
@@ -351,7 +351,9 @@ module Crystal
       return if @debug.none?
       in_alloca_block do
         vars.each do |name, var|
-          llvm_var = context.vars[name]
+          # If a variable is deduced to have type `NoReturn` it might not be
+          # allocated at all
+          next unless (llvm_var = context.vars[name]?)
```

Comment:
This is totally optional, but I prefer the assignment and the condition to be split. Then the comment is only relevant to the condition, not the assignment.